modules/toDoListFunctionlity.js

```diff
@@ -0,0 +1,143 @@
+      // Adding event to checkbox
+      this.checkBoxStatusEvent(item, checkboxElement, pElement);
+      // remove Button
+      this.removeItem(this.toDoListArray, removeBtn, indexPosition, liElement);
+      // 1- Append New Element to Parent node which is li.
+      // 2- Append the li to ul.
+      liElement.append(checkboxElement, pElement, removeBtn);
+      this.parentUl.appendChild(liElement);
+    });
+  };
+
+  editeInputText = (inputTextAddress, arrayDescription) => {
+    inputTextAddress.addEventListener('focusout', () => {
+      if (inputTextAddress.value !== '') {
+        arrayDescription.description = inputTextAddress.value;
+        localStorage.setItem('toDoListItems', JSON.stringify(this.toDoListArray));
+      } else {
+        inputTextAddress.value = arrayDescription.description;
+      }
+    });
+  }
+
+  checkBoxStatusOnLoad = (updateStatusArray, checkboxAddress, textInputAddress) => {
+    if (updateStatusArray.completed === true) {
+      checkboxAddress.setAttribute('class', 'completed-task');
+      checkboxAddress.setAttribute('checked', 'checked');
+      textInputAddress.style.textDecoration = 'line-through';
+      textInputAddress.style.color = '#999';
+    } else {
+      checkboxAddress.removeAttribute('class');
+      checkboxAddress.removeAttribute('checked');
+      textInputAddress.style.textDecoration = 'none';
+      textInputAddress.style.color = '#000';
+      updateStatusArray.completed = false;
+    }
+  }
+
+  checkBoxStatusEvent = (updateStatusArray, checkboxAddress, textInputAddress) => {
+    checkboxAddress.addEventListener('change', () => {
+      if (updateStatusArray.completed === false) {
+        checkboxAddress.setAttribute('class', 'completed-task');
+        checkboxAddress.setAttribute('checked', 'checked');
+        textInputAddress.style.textDecoration = 'line-through';
+        textInputAddress.style.color = '#999';
+        updateStatusArray.completed = true;
+        localStorage.setItem('toDoListItems', JSON.stringify(this.toDoListArray));
+      } else {
+        checkboxAddress.removeAttribute('checked');
+        checkboxAddress.removeAttribute('class');
+        textInputAddress.style.textDecoration = 'none';
+        textInputAddress.style.color = '#000';
+        updateStatusArray.completed = false;
+        localStorage.setItem('toDoListItems', JSON.stringify(this.toDoListArray));
+      }
+    });
+  }
+
+  removeItem = (arrayDoToList, btnAddress, itemIndexPosition, liAddress) => {
+    // Use to create event Listener for each passen btn.
+    btnAddress.addEventListener('click', () => {
+      this.toDoListArray.splice(itemIndexPosition, 1);
+      // reassigne index to array items
+      arrayDoToList.forEach((element, elementIndex) => {
+        element.index = elementIndex;
+      });
+      localStorage.setItem('toDoListItems', JSON.stringify(this.toDoListArray));
+      liAddress.remove();
+      window.location.reload();
+    });
+  }
```

Comment:
- [x]It would be better if you use the e.target to get the new value. It will be short code and more understandable.

Reply:
Hi @soesandarwin2201 
Thanks a lot for your great comment and good feedback I will make the changes right away. ✔✔✔